webapp/test_runs_handler.go

```diff
@@ -17,36 +17,49 @@ func testRunsHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "Only GET is supported.", http.StatusMethodNotAllowed)
 		return
 	}
-	ctx := shared.NewAppEngineContext(r)
-	aeAPI := shared.NewAppEngineAPI(ctx)
-	testRunFilter, err := shared.ParseTestRunFilterParams(r.URL.Query())
+	filter, err := parseTestRunsUIFilter(r)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
-	// Get runs from a month ago, onward, by default.
-	if testRunFilter.IsDefaultQuery() {
-		aWeekAgo := time.Now().Truncate(time.Hour*24).AddDate(0, 0, -7)
-		testRunFilter.From = &aWeekAgo
-		if aeAPI.IsFeatureEnabled("masterRunsOnly") {
-			testRunFilter = testRunFilter.MasterOnly()
-		}
-	} else if testRunFilter.MaxCount == nil {
-		oneHundred := 100
-		testRunFilter.MaxCount = &oneHundred
+	if err := templates.ExecuteTemplate(w, "test-runs.html", filter); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
 	}
+}
 
-	filter := convertTestRunUIFilter(testRunFilter)
+// parseTestRunsUIFilter parses the standard TestRunFilter, as well as the extra
+// pr param.
+func parseTestRunsUIFilter(r *http.Request) (filter testRunUIFilter, err error) {
+	q := r.URL.Query()
+	testRunFilter, err := shared.ParseTestRunFilterParams(q)
+	if err != nil {
+		return filter, err
+	}
 
-	data := struct {
-		Filter testRunUIFilter
-	}{
-		Filter: filter,
+	pr, err := shared.ParsePRParam(q)
+	if err != nil {
+		return filter, err
 	}
 
-	if err := templates.ExecuteTemplate(w, "test-runs.html", data); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+	if pr != nil {
+		filter.PR = pr
+	} else {
+		if testRunFilter.IsDefaultQuery() {
+			// Get runs from a month ago, onward, by default.
```

Comment:
This comment doesn't match the code, it seems.

Reply:
Done.